lib/vagrant-service-manager/action/restart_service.rb

```diff
@@ -0,0 +1,34 @@
+module Vagrant
+  module ServiceManager
+    module Action
+      class RestartService
+        def initialize(app, env)
+          @app = app
+          @machine = env[:machine]
+          @ui = env[:ui]
+          @providers = @machine.config.servicemanager.providers.split(',').map(&:chomp)
+        end
+
+        def execute_command(command, provider)
+           @machine.communicate.execute(command) do |type, data|
+            if type == :stderr
+              @ui.error(data)
+            else
+              @ui.info("# Restarted #{provider} service.")
+            end
+          end
+        end
+
+        def call(env)
+          @providers.each do |provider|
+            command = "sudo systemctl restart #{provider}"
```

Comment:
The benefit of a more modular approach, which I believe is used by atomic app too, is that it is easier for those unfamiliar with the whole code base to add a new service.